packages/manager/apps/catalog-revamp/src/utils/utils.ts

```diff
@@ -0,0 +1,140 @@
+export type Product = {
+  id: number;
+  name: string;
+  description: string;
+  lang: string;
+  categories: string[];
+  url: string;
+  regionTags: string[];
+  productName: string;
+  order: string;
+  universe: string;
+  category: string;
+};
+
+type ProductKey = 'universe' | 'category' | 'name';
+
+export const groupItemsByUniverse = (itemsToGroup: Product[]) => {
+  return itemsToGroup.reduce<Record<string, Product[]>>((acc, item) => {
+    if (!acc[item.universe]) acc[item.universe] = [];
+    acc[item.universe].push(item);
+    return acc;
+  }, {});
+};
+
+export const filterByEnum = (
+  product: Product,
+  query: string[],
+  property: ProductKey,
+) => {
+  return query.length === 0 || query.includes(product[property]);
+};
+
+export const getAvailableCategories = (
+  products: Product[],
+  selectedUniverses: string[],
+) => {
+  const itemsFiltered = products.filter((product) =>
+    filterByEnum(product, selectedUniverses, 'universe'),
+  );
+  const uniqueCategories = [
+    ...new Set(itemsFiltered.map((item) => item.category)),
+  ];
+  return uniqueCategories.sort((a, b) =>
+    a.localeCompare(b, undefined, { sensitivity: 'base' }),
+  );
+};
+
+export const matchCategories = (
+  product: Product,
+  selectedCategories: string[],
+) => {
+  return filterByEnum(product, selectedCategories, 'category');
+};
+
+export const matchUniverses = (
+  product: Product,
+  selectedUniverses: string[],
+) => {
+  return filterByEnum(product, selectedUniverses, 'universe');
+};
```

Comment:
those functions are not really usefull : just use filterByProperty directly

Reply:
Done